In [1]:
import pandas as pd


In [2]:
import zipfile

with zipfile.ZipFile("higher+education+students+performance+evaluation.zip", "r") as zip_ref:
    zip_ref.extractall()

In [3]:
df = pd.read_csv("DATA (1).csv")


In [4]:
df.shape

(145, 33)

A. How many rows and columns are contained in the dataset?

The dataset contains 145 rows and 33 columns.

In [5]:
df.shape

(145, 33)

B.What datatypes are contained in the dataset?

In [6]:
df.dtypes


,0
STUDENT ID,object
1,int64
2,int64
3,int64
4,int64
5,int64
6,int64
7,int64
8,int64
9,int64


The dataset contains two datatypes: object and int64. The STUDENT ID column has an object datatype, while the remaining 32 columns have an int64 datatype.

C. Is the dataset complete? i.e. no missing values

In [7]:
df.isnull().sum()

,0
STUDENT ID,0
1,0
2,0
3,0
4,0
5,0
6,0
7,0
8,0
9,0


The dataset is complete because all columns have 0 missing values.
Therefore, the dataset contains no missing values.

In [8]:
df.head(15)

,STUDENT ID,1,2,3,4,5,6,7,8,9,...,23,24,25,26,27,28,29,30,COURSE ID,GRADE
0,STUDENT1,2,2,3,3,1,2,2,1,1,...,1,1,3,2,1,2,1,1,1,1
1,STUDENT2,2,2,3,3,1,2,2,1,1,...,1,1,3,2,3,2,2,3,1,1
2,STUDENT3,2,2,2,3,2,2,2,2,4,...,1,1,2,2,1,1,2,2,1,1
3,STUDENT4,1,1,1,3,1,2,1,2,1,...,1,2,3,2,2,1,3,2,1,1
4,STUDENT5,2,2,1,3,2,2,1,3,1,...,2,1,2,2,2,1,2,2,1,1
5,STUDENT6,2,2,2,3,2,2,2,2,1,...,1,1,1,2,1,2,4,4,1,2
6,STUDENT7,1,2,2,4,2,2,2,1,1,...,1,1,3,3,3,3,4,4,1,5
7,STUDENT8,1,1,2,3,1,1,1,2,2,...,3,1,3,2,2,1,1,1,1,2
8,STUDENT9,2,1,3,3,2,1,1,1,1,...,1,1,3,2,2,2,4,3,1,5
9,STUDENT10,2,1,2,3,2,2,1,3,4,...,1,1,2,2,2,2,1,2,1,0


In [9]:
df.tail(20)

,STUDENT ID,1,2,3,4,5,6,7,8,9,...,23,24,25,26,27,28,29,30,COURSE ID,GRADE
125,STUDENT126,1,1,2,5,1,1,2,1,1,...,1,2,3,2,3,1,1,3,9,2
126,STUDENT127,1,1,1,4,1,1,1,3,2,...,2,1,3,3,3,1,2,2,9,3
127,STUDENT128,1,1,2,4,2,2,2,1,4,...,1,1,3,2,2,1,2,2,9,1
128,STUDENT129,1,1,2,4,2,1,1,1,4,...,1,1,3,2,2,1,2,2,9,0
129,STUDENT130,1,1,2,3,1,1,2,1,1,...,2,2,3,2,3,2,2,3,9,3
130,STUDENT131,1,1,2,3,1,1,2,1,1,...,1,2,3,1,3,2,2,3,9,1
131,STUDENT132,1,1,1,5,2,1,2,1,2,...,1,2,3,2,3,1,5,3,9,4
132,STUDENT133,1,1,1,5,2,1,2,1,1,...,2,1,3,2,3,1,5,4,9,3
133,STUDENT134,1,1,2,5,2,2,1,1,1,...,1,1,3,3,3,2,5,3,9,3
134,STUDENT135,1,1,2,4,2,1,2,1,2,...,1,1,2,1,2,1,2,3,9,1


In [10]:
df_sample = pd.concat([df.head(15), df.tail(20)])

In [11]:
df_sample

,STUDENT ID,1,2,3,4,5,6,7,8,9,...,23,24,25,26,27,28,29,30,COURSE ID,GRADE
0,STUDENT1,2,2,3,3,1,2,2,1,1,...,1,1,3,2,1,2,1,1,1,1
1,STUDENT2,2,2,3,3,1,2,2,1,1,...,1,1,3,2,3,2,2,3,1,1
2,STUDENT3,2,2,2,3,2,2,2,2,4,...,1,1,2,2,1,1,2,2,1,1
3,STUDENT4,1,1,1,3,1,2,1,2,1,...,1,2,3,2,2,1,3,2,1,1
4,STUDENT5,2,2,1,3,2,2,1,3,1,...,2,1,2,2,2,1,2,2,1,1
5,STUDENT6,2,2,2,3,2,2,2,2,1,...,1,1,1,2,1,2,4,4,1,2
6,STUDENT7,1,2,2,4,2,2,2,1,1,...,1,1,3,3,3,3,4,4,1,5
7,STUDENT8,1,1,2,3,1,1,1,2,2,...,3,1,3,2,2,1,1,1,1,2
8,STUDENT9,2,1,3,3,2,1,1,1,1,...,1,1,3,2,2,2,4,3,1,5
9,STUDENT10,2,1,2,3,2,2,1,3,4,...,1,1,2,2,2,2,1,2,1,0
